pkg/providers/amifamily/resolver.go

```diff
@@ -222,31 +252,38 @@ func (r DefaultResolver) resolveLaunchTemplate(nodeClass *v1.EC2NodeClass, nodeC
 	}); !found {
 		taints = append(taints, karpv1.UnregisteredNoExecuteTaint)
 	}
-
-	resolved := &LaunchTemplate{
-		Options: options,
-		UserData: amiFamily.UserData(
-			r.defaultClusterDNS(options, kubeletConfig),
-			taints,
-			options.Labels,
-			options.CABundle,
-			instanceTypes,
-			nodeClass.Spec.UserData,
-			options.InstanceStorePolicy,
-		),
-		BlockDeviceMappings: nodeClass.Spec.BlockDeviceMappings,
-		MetadataOptions:     nodeClass.Spec.MetadataOptions,
-		DetailedMonitoring:  aws.ToBool(nodeClass.Spec.DetailedMonitoring),
-		AMIID:               amiID,
-		InstanceTypes:       instanceTypes,
-		EFACount:            efaCount,
-		CapacityType:        capacityType,
-	}
-	if len(resolved.BlockDeviceMappings) == 0 {
-		resolved.BlockDeviceMappings = amiFamily.DefaultBlockDeviceMappings()
-	}
-	if resolved.MetadataOptions == nil {
-		resolved.MetadataOptions = amiFamily.DefaultMetadataOptions()
+	// If no reservation IDs are provided, insert an empty string so the end result is a single launch template with no
+	// associated capacity reservation.
+	if len(capacityReservationIDs) == 0 {
+		capacityReservationIDs = append(capacityReservationIDs, "")
```

Comment:
I get what you're doing here but (to me) this is confusing -- what if you just created the launch template template above and just returned that if the length of the `capacityReservationIDs` was 0 and then copied it and added the reservationID to it if you have to iterate through a list?

Reply:
Agreed to leave this as is for now, we really would want to do a DeepCopy of the `LaunchTemplate` struct but we're having trouble with the `deepcopy-gen` tool since it contains the `bootstrap.Bootstrapper` interface.